packages/astro/src/core/add/index.ts

```diff
@@ -407,7 +407,7 @@ async function getInstallIntegrationsCommand({
 		.map<[string, string | null][]>((i) => [[i.packageName, null], ...i.dependencies])
 		.flat(1)
 		.filter((dep, i, arr) => arr.findIndex((d) => d[0] === dep[0]) === i)
-		.map(([name, version]) => (version === null ? name : `${name}@${version}`))
+		.map(([name, version]) => (version === null ? name : `${name}@${version.split(' ').pop()}`))
```

Comment:
_slightly_ paranoid this could break on `18.1.0|| 17.0.2` for example. Non-blocking, but could be regexed like so!  This allows any number of spaces between the `||` separator

```suggestion
		.map(([name, version]) => (version === null ? name : `${name}@${version.split(/\s*\|\|\s*/).pop()}`))
```

Reply:
I'm fine with this as well. Like I mentioned, the only absolutely 100% correct thing here is to bring in a some sort of semver parsing library, which I think is overkill for something that we control ourselves (remember this is a semver value coming from the integration that we author!)

Reply:
I'd go with Ben's suggestion and then it LGTM